lib/Partitioner/Partitioner.cpp

```diff
@@ -468,6 +468,7 @@ void Partitioner::doPartitioning(Function *F, NodeToFunctionMap &mapping) {
     }
 
     // Link subF to its parents.
+    std::set<Function *> parents;
```

Comment:
how hard would be to exercise buggy behavior in a unit test?

Reply:
Actually, it won't show errors in the release mode. This may trigger the assertion that a placeholder is already registered.

Reply:
Can we add a test for that? (debug mode failure would be enough)

Reply:
I had one. Will added it later :)